src/test/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfigTest.java

```diff
@@ -0,0 +1,159 @@
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+/**
+ * Tests for SalesforceStreamingSourceConfig
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({SalesforceStreamingSourceConfig.class, SalesforceStreamingSource.class, SObjectDescriptor.class,
+        SalesforceQueryUtil.class, SObjectFilterDescriptor.class})
+public class SalesforceStreamingSourceConfigTest {
+
+    SalesforceStreamingSourceConfig actualSalesforceStreamingSourceConfig;
+    OAuthInfo oAuthInfo;
+
+    @Before
+    public void setUp() {
+
+        oAuthInfo = new OAuthInfo("token", "https://d5j000001ufckeay.lightning.force.com");
+
+        actualSalesforceStreamingSourceConfig = new SalesforceStreamingSourceConfig(
+                "Reference Name", "Consumer Key", "Consumer Secret",
+                "username", "password", "https://example.org/example",
+                "Push Topic Name", "S Object Name", "token", oAuthInfo);
+    }
+
+    @Test
+    public void testConstructor() {
+        assertEquals("Push Topic Name", actualSalesforceStreamingSourceConfig.getPushTopicName());
+        assertNull(actualSalesforceStreamingSourceConfig.getPushTopicNotifyForFields());
+        assertNull(actualSalesforceStreamingSourceConfig.getPushTopicQuery());
+        assertEquals("Reference Name", actualSalesforceStreamingSourceConfig.referenceName);
+        assertEquals("Consumer Key", actualSalesforceStreamingSourceConfig.getConsumerKey());
+        assertEquals("https://example.org/example", actualSalesforceStreamingSourceConfig.getLoginUrl());
+        assertEquals("Push Topic Name", actualSalesforceStreamingSourceConfig.getPushTopicName());
+        assertSame(oAuthInfo, actualSalesforceStreamingSourceConfig.getOAuthInfo());
+        assertEquals("Consumer Secret", actualSalesforceStreamingSourceConfig.getConsumerSecret());
+        assertEquals("passwordtoken", actualSalesforceStreamingSourceConfig.getPassword());
+        assertEquals("username", actualSalesforceStreamingSourceConfig.getUsername());
+    }
+
+    @Test
+    public void testFetchPushTopicByNameInvalidFieldName()
+            throws ConnectionException {
+        PartnerConnection partnerConnection = mock(PartnerConnection.class);
+        try {
+            SalesforceStreamingSourceConfig.fetchPushTopicByName(partnerConnection, ",");
+        } catch (IllegalArgumentException e) {
+            assertEquals("Push topic name ',' can only contain latin letters.", e.getMessage());
+        }
```

Comment:
fail if exception is not thrown

Reply:
Failure Assertion added if no exception thrown

Reply:
I don't see this added. Did you forget to push a commit?

Reply:
The failure assertion is added here. https://github.com/data-integrations/salesforce/blob/70835f290791abcf53205b6570a28e17c6746436/src/test/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfigTest.java#L83